internal/backend/telemetry/telemetry.go

```diff
@@ -0,0 +1,120 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	api "go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"go.uber.org/zap"
+)
+
+func toMetricAttrs(attrs map[string]string) []attribute.KeyValue {
+	var result []attribute.KeyValue
+	for k, v := range attrs {
+		result = append(result, attribute.String(k, v))
+	}
+	return result
+}
+
+type Metric interface {
+	Update(value int64, attrs map[string]string)
+}
+
+type Counter struct {
+	counter api.Int64Counter
+}
+
+type UpDownCounter struct {
+	counter api.Int64UpDownCounter
+}
+
+func (m Counter) Update(value int64, attrs map[string]string) {
+	// REVIEW: should we check/report if we get a negative value
+	m.counter.Add(context.Background(), value, api.WithAttributes(toMetricAttrs(attrs)...))
```

Comment:
why background context instead of passing context ?

Reply:
which context you want to pass and why?
we are initializing object of opentelemetry and I don't think we should pass there any context except Background

Reply:
There is always a context from the user or the workflow, no reason not passing it

Reply:
there is no reason to pass it as well, unless we want to add tags from the context. 
So meanwhile the simplest option is skip passing context, I think.

Reply:
We need to pass a proper context, never `Background()`. I understand why you wouldn't want to pass a context from the workflow, but you could at least limit the context passed to this function. By limiting, I mean doing `WithTimeout` with a small value (1s?). If there is any error - log. That way this will not be stuck for a meaningful amount of time.